shap/explainers/_kernel.py

```diff
@@ -397,9 +397,12 @@ def explain(self, incoming_instance, **kwargs):
 
     @staticmethod
     def not_equal(i, j):
-        if isinstance(i, str) or isinstance(j, str):
-            return 0 if i == j else 1
-        return 0 if np.isclose(i, j, equal_nan=True) else 1
+        try:
+            if isinstance(i, str) or isinstance(j, str):
+                return 0 if i == j else 1
+            return 0 if np.isclose(i, j, equal_nan=True) else 1
+        except Exception:
+            return int(i != j)
```

Comment:
Shouldn't you have an explicit case for timestamp types instead of a try/except?  eg check isinstance(i, datetime) or similar for timestamp and then return i != j?
Also, can you add a test for this?

Reply:
@imatiach-msft I would like to support all kinds of types besides the `np.isclose` supported types and string type.
I can improve code to remove try...except.

Reply:
this looks much better!